tests/unit/moduleapi/reply.tcl

```diff
@@ -28,6 +28,29 @@ start_server {tags {"modules"}} {
             assert_equal 3.141 [r rw.double 3.141]
         }
 
+        test "RESP$proto: RM_ReplyWithDouble: inf" {
+            if {$proto == 2} {
+                assert_equal "inf" [r rw.double inf]
+                assert_equal "-inf" [r rw.double -inf]
+            } else {
+                assert_equal Inf [r rw.double inf]
+                assert_equal -Inf [r rw.double -inf]
+            }
+        }
+
+        test "RESP$proto: RM_ReplyWithDouble: NaN" {
+            if {$proto == 2} {
+                assert_equal "-nan" [r rw.double 0 0]
+                assert_equal "nan" [r rw.double]
```

Comment:
Negative NaN??? What is that...

Reply:
apparently that's what you get from redis (or actually both glibc and the alternative we added in #10587).
so either we "fix" it with a breaking change, or document it.

Reply:
Fine then. Apparently the sign bit in a NaN is usually ignored but not always... Wikipedia:

![image](https://user-images.githubusercontent.com/273886/200399297-51629c10-654e-41cc-8004-c6cfe5cb4e9b.png)

Reply:
ohh, now i see the Negative NaN also break ours extra-cli
https://github.com/redis/redis-extra-ci/actions/runs/3457946091/jobs/5771873230#step:8:846
```
*** [err]: RESP2: RM_ReplyWithDouble: NaN in tests/unit/moduleapi/reply.tcl
Expected '-nan' to be equal to 'nan' (context: type eval line 3 cmd {assert_equal "-nan" [r rw.double 0 0]} proc ::test)
*** [err]: RESP3: RM_ReplyWithDouble: NaN in tests/unit/moduleapi/reply.tcl
Expected ',-nan' to be equal to ',nan' (context: type eval line 8 cmd {assert_equal ",-nan" [r rw.double 0 0]} proc ::test)
```

Reply:
i'm careless.. i should have tested this on Mac and ARM before merging.

Reply:
let's do pattern matching to search any kind of `nan`, ignoring character case and sign.
@enjoy-binbin can you make a PR?

Reply:
make a quick PR: #11506